### PR TITLE
[OSDOCS-9251]:  Remove references to 'machineconfigpools' from NTO on ROSA with HCP docs

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.14+
Issue:

https://issues.redhat.com/browse/OSDOCS-9251
Link to docs preview:

https://70125--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_hcp/rosa-tuning-config

Reviewers: You can use this link to confirm that nothing was changed in the OCP docs, as this doc update should not affect those docs: 
https://70125--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/using-node-tuning-operator#custom-tuning-specification_node-tuning-operator
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/custom-tuning-specification.adoc [hunk 1]
```diff
@@ -148,9 +147,7 @@ ifdef::rosa-hcp-tuning[]
 ----
 <1> A TuneD profile to apply on a match. For example `tuned_profile_1`.
 <2> Profile ordering priority. Lower numbers mean higher priority (`0` is the highest priority).
-<3> Optional: A dictionary of key-value pairs `MachineConfig` labels. The keys must be unique.
-<4> If omitted, profile match is assumed unless a profile with a higher priority matches first or `machineConfigLabels` is set.
-<5> The label for the profile matched items.
+<3> The label for the profile matched items.
```
Comment (dagrayvid) [inline modules/custom-tuning-specification.adoc:150]: I think we can keep the description of the <match> section above and reword this to just remove the "or `machineConfigLabels` is set." part.

```suggestion
<3> If omitted, profile match is assumed unless a profile with a higher priority matches first
<4> The label for the profile matched items.
```

File: modules/custom-tuning-specification.adoc [hunk 2]
```diff
@@ -134,12 +134,11 @@ ifdef::rosa-hcp-tuning[]
   "recommend": [
     {
       "profile": <tuned_profile_name>, <1>
```
Comment (dagrayvid) [inline modules/custom-tuning-specification.adoc:136]: All of these changes to remove references to machineConfigLabels only apply to HCP, so they should be copied to a separate section behind an ifdef. We don't want to change these docs in the standalone OCP docs.

File: modules/custom-tuning-specification.adoc [hunk 3]
```diff
@@ -290,6 +289,7 @@ spec:
 ----
 endif::rosa-hcp-tuning[]
 ifdef::rosa-hcp-tuning[]
+.Example: Machine pool based matching
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/custom-tuning-specification.adoc:292]: > Why have we suddently started using JSON vs. YAML ? YAML is used by all regular OCP docs.

Hello. While I can't answer that question directly as I was never part of that decision, I can tell you that I have seen JSON being used from the time I started here (12/22) so not sure how new it is/when it was implemented.

File: modules/custom-tuning-specification.adoc (hunk 2 above)
Comment (mletalie) [reply modules/custom-tuning-specification.adoc:136]: Checked preview of topic in OCP with current (published) topic in OCP and they match, so conditionals have separated the changes (i.e. current OCP docs will not be affected by changes).

File: modules/custom-tuning-specification.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmencak) [reply modules/custom-tuning-specification.adoc:292]: I suspect this will likely need more work. JSON is not as readable as YAML. The configuration ĺooks very strange in this way.  I'm planning to set-up a ROSA with HCP cluster to test this. Who is normally testing this on ROSA HCP? Perhaps we could arrange to share a test environment.

[the remark this answers is not in the document]
Comment (mletalie) [reply modules/custom-tuning-specification.adoc:292]: > I suspect this will likely need more work. JSON is not as readable as YAML. The configuration ĺooks very strange in this way. I'm planning to set-up a ROSA with HCP cluster to test this. Who is normally testing this on ROSA HCP? Perhaps we could arrange to share a test environment.

Hello @jmencak, understood. I have opened another ticket to address these additional concerns as based on the recent Slack discussions on this issue (https://issues.redhat.com/browse/OSDOCS-9296). I can reach out to our QE  "go-to" person and ask about sharing an environment. 
Sorry I could not be of more help.

[the remark this answers is not in the document]
Comment (jmencak) [reply modules/custom-tuning-specification.adoc:292]: It could be the the json format is required. I don't know yet, just starting to get familiar with this.  But just trying to create a simple basic tuning configuration, I'm getting:

```
rosa create tuning-config -c <my cluster> --name my-tuning-config --spec-path ./tuning-config.json
E: Expected a valid TuneD spec file: invalid character '\n' in string literal
```

It would be nice who the QE for this is, so that I could work with them. Thank you.

[the remark this answers is not in the document]
Comment (mletalie) [reply modules/custom-tuning-specification.adoc:292]: Roger that, @jmencak. I have sent a Slack message to @xueli181114 to see if she can assist with that info. Hopefully we can get an answer soon. Thanks!

[the remark this answers is not in the document]
Comment (xueli181114) [reply modules/custom-tuning-specification.adoc:292]: @radtriste will you look at this?

[the remark this answers is not in the document]
Comment (jmencak) [reply modules/custom-tuning-specification.adoc:292]: Started a thread on this: https://redhat-internal.slack.com/archives/CHWDU0CP9/p1705484998082819

[the remark this answers is not in the document]
Comment (jmencak) [reply modules/custom-tuning-specification.adoc:292]: I can no longer reproduce the issue. Not sure what the problem was, but with the config:

```
{
  "profile": [
    {
      "data": "[main]\nsummary=Custom OpenShift profile\ninclude=openshift-node\n[sysctl]\nvm.dirty_ratio=\"55\"",
      "name": "tuned-1-profile"
    }
  ],
  "recommend": [
    {
      "priority": 20,
      "profile": "tuned-1-profile"
    }
  ]
}
```

The `tuning-config` is created without any issues.  Appologies for the noise.  What is unfortunate though, is that the rosa client only supports JSON as the input format for the configuration. [Hopefully that changes in the future](https://redhat-internal.slack.com/archives/CHWDU0CP9/p1705494874344759?thread_ts=1705484998.082819&cid=CHWDU0CP9).

@mletalie , thank you for your help and please feel free to close my JSON comment as resolved.

**Comment by mletalie** [comment]
> Thank you for the work so far, @mletalie . I can stil stilll see a few potential issues with the docs.
> 
> 1. On Rosa HCP, I don't think there is any support for `PerformanceProfiles`.  I believe we need more `ifdef`s to exclude them.  Most likely the whole section that starts:
> 
> "The Node Tuning Operator uses the Performance Profile controller" ... and ends " if you are using performance profiles." should be removed by an `ifdef`.
> 
> 2. The entire section "Management state" should also be `ifdef`ed out. The operator management state is set by editing the default Tuned CR on regular OCP, but I don't see a way to do this in Rosa HCP.
> 
> We can consult the above with the QE or the "rosa" CLI developers to be sure. Please include them for the confirmation.

Hello @jmencak, 
Thanks for taking a look at this and the comments, info, etc. It is appreciated. What I am doing is using this ticket  https://issues.redhat.com/browse/OSDOCS-9296 to track further changes that need to be made to the ROSA HCP NTO docs, as it is clear that quite a few additional changes need to be made outside the original scope of OSDOCS-9251.
I will update that ticket with these (and all the other) comments. Thanks again!

**Review by skrthomas — COMMENTED** [review]
These updates /lgtm. Nice work!!

**Comment by mletalie** [comment]
> These updates /lgtm. Nice work!!
Thanks!

File: modules/custom-tuning-specification.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/custom-tuning-specification.adoc:292]: Thanks @jmencak, Appreciate the heads up!
